flink-core/src/main/java/org/apache/flink/configuration/TraceOptions.java

```diff
@@ -67,6 +82,19 @@ public static Configuration forReporter(Configuration configuration, String repo
                                     + " any of the names in the list will be started. Otherwise, all reporters that could be found in"
                                     + " the configuration will be started.");
 
+    /** The detail level for reporting checkpoint spans. */
+    public static final ConfigOption<TraceOptions.CheckpointSpanDetailLevel>
+            CHECKPOINT_SPAN_DETAIL_LEVEL =
+                    key("traces.checkpoint.span-detail-level")
+                            .enumType(TraceOptions.CheckpointSpanDetailLevel.class)
+                            .defaultValue(CheckpointSpanDetailLevel.SPANS_PER_CHECKPOINT)
+                            .withDescription(
+                                    "Detail level for reporting checkpoint spans. Possible values:\n"
+                                            + "- SPAN_PER_CHECKPOINT (default): Single span per checkpoint. Aggregated sum/max for submetrics from all tasks and subtasks per checkpoint\n"
```

Comment:
```suggestion
                                            + "- SPANS_PER_CHECKPOINT (default): Single span per checkpoint. Aggregated sum/max for submetrics from all tasks and subtasks per checkpoint\n"
```
Same for a few more times across the description

Reply:
I have also re-used constants to avoid problems with names getting out of sync.